### DISCUSSION TITLE
PaddleOCR on Apple Silicon

### DISCUSSION THREAD

**Discussion by lukasugar** [PaddlePaddle/PaddleOCR#12795] [Q&A] [answered]
It's noted in older issues that new PaddleOCR versions don't support Apple Silicon: https://github.com/PaddlePaddle/PaddleOCR/issues/4838

Can this be sovled? 🙏 

The new PaddleOCR versions support better models, it's a pity we can't use them.

Because of dependency to `PyMuPDF<1.21.0`, I end up with these package versions:
```
paddlepaddle==2.5.0
paddleocr>=2.0.1
PyMuPDF==1.21.1
numpy==1.23.5
```

Bacause of old PyMuPDF, if I want to use `setuptools` to build my project, I need to go all the way back to these versions:
```
setuptools==57.5.0,
Mako==1.2.4
```

Is adding support for Apple Silicon for newer versions on the roadmap? Any ETAs?

Or is there a way to get the newer PaddleOCR packages work on Apple Silicon?

**Comment by JayveeHe** [comment, marked as answer]
Same issue here, python 3.10.8, M1 Max

solved by:
1. git clone and switch to 'release/2.7' branch
2. change 'PyMuPDF<1.21.0' to 'PyMuPDF' in requirements.txt
3. remove all '*.pyc'  files
4. pip install 'PyMuPDF==1.21.0'
5. python setup.py install - to install PaddleOCR from source.

Haven't tried PDF-related features, but OCR for images is fine.

**Comment by SUPERustam** [comment]
I also have same issue. @JayveeHe I tried your method to solve problem with m-chip on Python 3.9 and Python 3.11, but still cann't run PaddleOCR. 
Where located *.pyc files in project? Maybe I do something wrong? Did you this instruction in native or Rosseta2 emulator?

**Comment by hrokholskyi** [comment]
Using paddlepaddle==3.0.0b2 have resolved this issue, hopefully it will come out of beta soon
